### PR TITLE
BFD-351: Revert breaking RestFul server init changes.

### PR DESCRIPTION
**Pull request by jzulim**
### Change Details
This change switches back to using the deprecated method of initializing a HAPI RestFul server. I tried to get fancy with V2 and swap this out. It ended up breaking the V2 servlet init, which did not show on IT's because we dont test V2. If V2 failed, V1 failed, thus breaking deployments.

### Acceptance Validation
I have now confirmed both servers (V1, V2) work locally and also deploy successfully to TEST.

### Feedback Requested
Should I attack this deprecated code now? I'm afraid this will drag out the release of this much-anticipated feature. I have marked TODO's and will open the backlog item. 

### External References

- [BFD-351](https://jira.cms.gov/browse/BFD-351)

### Security Implications

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/V1Server.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ private void configureServerInfoMetadata() {
 
     // Lightly customize the capability provider to set publisher name.
     ServerCapabilityStatementProvider capabilityStatementProvider =
-        new ServerCapabilityStatementProvider(this);
+        new ServerCapabilityStatementProvider();
```
Comment (jzulim) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/V1Server.java:69]: moved to no arg constructor per documentation.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/V2Server.java [hunk 2]
```diff
@@ -94,7 +94,8 @@ protected void initialize() throws ServletException {
      * global history.
      */
     List<Object> plainProviders = new ArrayList<>();
-    registerProvider(plainProviders);
```
Comment (jzulim) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/V2Server.java:97]: This change was breaking V2, which broke deployments. I reverted back the original and deprecated method.

**Review by cbrunefearless — APPROVED** [review]
LGTM

**Review by keithdadkins — APPROVED** [review]
LGTM

**Pull request merged**